src/systems/userinput/userinput.js

```diff
@@ -429,6 +436,14 @@ AFRAME.registerSystem("userinput", {
         xrSession.requestReferenceSpace("local-floor").then(referenceSpace => {
           this.xrReferenceSpace = referenceSpace;
         });
+        xrSession.addEventListener("end", e => {
+          const xrDevices = this.activeDevices.items.filter(d => d.gamepad && d.gamepad.isWebXRGamepad);
+          for (const device of xrDevices) {
+            if (device.gamepad && device.gamepad.isWebXRGamepad) {
+              gamepadDisconnected(device);
+            }
+          }
+        });
```

Comment:
Why filtering two times? One is `this.activeDevices.items.filter(d => d.gamepad && d.gamepad.isWebXRGamepad)` and another one is `if (device.gamepad && device.gamepad.isWebXRGamepad)`. I think we can drop the latter one.

Reply:
Ah yep good catch. I was originally just using the loop, but can't do that since the structure is being mutated so I added the pre-filter, forgot to remove the if.